third_party/move/move-model/src/builder/exp_builder.rs

```diff
@@ -2328,6 +2383,33 @@ impl<'env, 'translator, 'module_translator> ExpTranslator<'env, 'translator, 'mo
         }
     }
 
```

Comment:
It would be good to add some test cases where lambda params and return types involve some automatic type coercion (between `&mut` and `&`) due to subtyping:
- acceptable "mismatch" between expected and provided types (automatic type coercion makes it okay to have these mismatches), on the return type and param type of a lambda (the directionality of the acceptable mismatch is different for these)
- erroneous mismatch between expected and provided types on the return type and param types of a lambda.

Reply:
I've added `lambda_typed_widen.move` that tries to do something with that, but I can't think of any subtyping relationship we have other than `&` and `&mut`.  Any better ideas?

Reply:
I believe that is the only subtyping relationship we currently have.